### PR TITLE
Integrate LLVM at llvm/llvm-project@6a8ba31

### PR DESCRIPTION
**Pull request by marbre**
Updates LLVM usage to match llvm/llvm-project@6a8ba31.
Further updates the MLIR-HLO submodule to tensorflow/mlir-hlo@ef42c50.

Upstream split `std.splat` into `tensor.splat` and `vector.splat` with
commit llvm/llvm-project@6a8ba31. Hence, the tensor-to-emitc pass now
has to support converting `tensor.splat` to an `emitc.call`. The
corresponding reference implementation is moved from `emitc::standard`
to `emitc::tensor`. Furthermore, since no other std operations are
supported via conversions, all std related passes are cleaned out.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,17 +53,14 @@ The `emitc-opt` tool supports the following options:
 | `--convert-mhlo-region-ops-to-emitc `    | Convert MHLO operations containing regions to EmitC dialect.             |
 | `--convert-mhlo-to-emitc `               | Convert from MHLO dialect to EmitC dialect.                              |
 | `--convert-arith-to-emitc `              | Convert arith dialect to EmitC dialect, replacing IndexCastOp.           |
-| `--convert-std-to-emitc `                | Convert std dialect to EmitC dialect, replacing SplatOp.                 |
 | `--convert-tensor-to-emitc `             | Convert tensor dialect to EmitC dialect, replacing ExtractOp.            |
```
Comment (henri-gruender) [inline README.md:56]:
```suggestion
| `--convert-tensor-to-emitc `             | Convert tensor dialect to EmitC dialect.                                 |
```
Conversion now contains multiple ops

File: include/emitc/Dialect/EmitC/Conversion/Passes.td [hunk 2]
```diff
@@ -29,12 +29,6 @@ def ConvertArithToEmitC : Pass<"convert-arith-to-emitc", "FuncOp"> {
...
 }
 
-def ConvertStdToEmitC : Pass<"convert-std-to-emitc", "FuncOp"> {
-  let summary = "Convert std dialect to EmitC dialect, replacing SplatOp.";
-  let constructor = "createConvertStdToEmitCPass()";
-  let dependentDialects = ["EmitCDialect"];
-}
-
 def ConvertTensorToEmitC : Pass<"convert-tensor-to-emitc", "FuncOp"> {
   let summary = "Convert tensor dialect to EmitC dialect, replacing ExtractOp.";
```
Comment (henri-gruender) [inline include/emitc/Dialect/EmitC/Conversion/Passes.td:33]:
```suggestion
  let summary = "Convert tensor dialect to EmitC dialect.";
```
Same as above

File: reference-implementation/unittests/tensor.cpp [hunk 3]
```diff
@@ -58,4 +60,35 @@ TEST(tensor, extract) {
   }
 }
 
+TEST(tensor, splat) {
```
Comment (henri-gruender) [inline reference-implementation/unittests/tensor.cpp:63]: Code was changed in #262

**Review by henri-gruender — CHANGES_REQUESTED** [review]
LGTM, except for the unittest and tensor-to-emitc summary

File: README.md (hunk 1 above)
Comment (marbre) [reply README.md]: Good fetch!

File: reference-implementation/unittests/tensor.cpp (hunk 3 above)
Comment (marbre) [reply reference-implementation/unittests/tensor.cpp:63]: I was aware of that, but seems I was on an out-of-date branch. Thanks for pointing me to this.

**Review by henri-gruender — APPROVED** [review]

**Pull request merged**